File: code_samples/go/web_development_rest_api.go
package main

import (
    "fmt"
)

// Web Development: Rest Api
// AI/ML Training Sample

type RestApi struct {
    Data string
}

func NewRestApi() *RestApi {
    return &RestApi{
        Data: "",
    }
}

func (s *RestApi) Process(input string) {
    s.Data = input
}

func (s *RestApi) Validate() bool {
    return len(s.Data) > 0
}

func (s *RestApi) GetData() string {
    return s.Data
}

func main() {
    instance := NewRestApi()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}

File: code_samples/haskell/web_development_rest_api.hs
-- Web Development: Rest Api
-- AI/ML Training Sample

module RestApi where

data RestApi = RestApi {
    getData :: String
} deriving (Show, Eq)

process :: RestApi -> String -> RestApi
process obj input = obj { getData = input }

validate :: RestApi -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = RestApi { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
